translator/source_code/icone_translator.py

```diff
@@ -353,6 +391,44 @@ def parse_incident(incident):
 
     return feature
 
+#function to validate the incident
+def validate_incident(incident):
+    
+    if not incident or (type(incident) != dict and type(incident) != OrderedDict):
+        logging.warning('incident is empty or has invalid type')
+        return False
+    
+    location = incident.get('location', {})
+    polyline = location.get('polyline')
```

Comment:
If we fail to get location from incident, this line will cause an exception. We check for location existence below so I assume this is not what we want.

Reply:
The second parameter to dictionary.get is what to return if the key does not exist, and for location we set the return value to an empty dictionary. Then, when we access location.get('polyline'), polyline will be None because 'polyline' is not present in the empty dictionary

Reply:
Ok that makes more sense. With this though, will we ever get anything out of checking the `if not location` below? This will always have a value (empty dictionary) right? If we run into the case where we don't find a location we then continue to grab all the other dictionary values before running into the empty polyline and returning false. It would make more sense not return the empty dictionary here and instead check for the null value before continuing to avoid the extra work.

Reply:
I just wanted to not have 30 if not *field* statements, I wanted to have 1 check at the end, if it was possible. Certainly there is no reason to check location, since polyline will always be None if location does not exist

Reply:
One check at the end certainly makes sense with the majority of these fields. Just the ones that are relied on by other fields could be checked separately, which here is really only location.

Reply:
I added a check for location right after retrieving it.